storagetransfer/transfer_from_s3_compatible_source.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetransfer
+
+import (
+	storagetransfer "cloud.google.com/go/storagetransfer/apiv1"
+	"context"
+	"fmt"
+	storagetransferpb "google.golang.org/genproto/googleapis/storagetransfer/v1"
+	"io"
+)
+
+// [START storagetransfer_transfer_from_s3_compatible_source]
+
+func transferFromS3CompatibleSource(w io.Writer, projectID string, sourceAgentPoolName string, sourceBucketName string, sourcePath string, gcsSinkBucket string, gcsPath string) (*storagetransferpb.TransferJob, error) {
+	// Your project id
+	// projectId := "my-project-id"
+
+	// The agent pool associated with the S3 compatible data source. If not provided, defaults to the default agent
+	// sourceAgentPoolName := "projects/my-project/agentPools/transfer_service_default"
+
+	// The S3 compatible bucket name to transfer data from
+	//sourceBucketName = "my-bucket-name"
+
+	// The S3 compatible path (object prefix) to transfer data from
+	//sourcePath = "path/to/data"
+
+	// The ID of the GCS bucket to transfer data to
+	//gcsSinkBucket = "my-sink-bucket"
+
+	// The GCS path (object prefix) to transfer data to
+	//gcsPath = "path/to/data"
+
+	// The S3 region of the source bucket
+	region := "us-east-1"
+
+	// The S3 compatible endpoint
+	endpoint := "us-east-1.example.com"
+
+	// The S3 compatible network protocol
```

Comment:
These options are a bit inscrutable but I assume the surrounding docs will clarify which to choose?

Reply:
The target audience here (people transferring from S3 compatible sources) should know what they mean, and yeah they're in the docs